Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -41,15 +41,22 @@ public final class ExpiryDateView: UIView {
     expiryDateView.update(style: style)
   }
   
-  private func updateErrorView(isHidden: Bool, text: String?){
-    style?.error?.isHidden = isHidden
-    style?.textfield.text = text ?? ""
-    expiryDateView.update(style: style)
+  private func updateErrorView(isHidden: Bool, text: String?, error: ValidationError.ExpiryDate? = nil){
+    defer {
+      style?.error?.isHidden = isHidden
+      style?.textfield.text = text ?? ""
+      expiryDateView.update(style: style)
+    }
+    guard let error = error else { return }
+    let errorText = error == .inThePast ?
+    Constants.LocalizationKeys.PaymentForm.ExpiryDate.Error.past :
+    Constants.LocalizationKeys.PaymentForm.ExpiryDate.Error.invalid
+    style?.error?.text = errorText
```

Comment:
Looking at this method it feels like the text is ignored when error exists. This would make me think 
> why do we need so many parameters if we purposely ignore some?

I wonder if it would be simpler and clearer if this method was still only receiving a text. I would even remove `isHidden`, as I think its a fair assumption that:
- you wouldn't show an error without text
- you wouldn't send text for error and not expect error to be shown

Code level, this would now be:
```swift
private func updateErrorView(text: String?) {
    let hideError = text?.isEmpty != false
    style?.error?.isHidden = hideError
    style?.textfield.text = text // Should be able to be optional ?
    expiryDateView.update(style: style)
}
```

Then to consumers where error was passed, they would now need to own converting an error to a message
```swift
let errorMessage = Constants.LocalizationKeys.PaymentForm.ExpiryDate.Error.past
updateErrorView(text: errorMessage)
```

More impact from this could be removing error cases (invalidYear, invalidMonth seem to have the same result for example)

Reply:
I now see that style has an errorText and a textField text, all update whilst updating error? When we _updateErrorView_, why the side effect of updating a textfield?

Reply:
`text` is in defer, it's not ignored

Reply:
I was getting confused on the `error.text` and `textField.text`. That to me is confusing as well. I still think we can improve on the code simplicity, but your unit tests do seem to show the behaviour works, so as far as strictly getting the work done, this seems to be ✅